src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -1624,15 +1620,11 @@ static int32_t guest_path_find_leaving_park(Peep* peep, uint8_t edges)
     if (chosenSpawn == 0xFF)
         return guest_path_find_aimless(peep, edges);
 
-    PeepSpawn* peepSpawn = &gPeepSpawns[chosenSpawn];
-
-    int16_t x = peepSpawn->x & 0xFFE0;
-    int16_t y = peepSpawn->y & 0xFFE0;
-    uint8_t z = peepSpawn->z / 8;
-    Direction direction = peepSpawn->direction;
+    const auto peepSpawnLoc = gPeepSpawns[chosenSpawn].ToTileStart();
+    Direction direction = peepSpawnLoc.direction;
 
-    gPeepPathFindGoalPosition = { x / 32, y / 32, z };
-    if (x == peep->NextLoc.x && y == peep->NextLoc.y)
+    gPeepPathFindGoalPosition = TileCoordsXYZ(peepSpawnLoc);
+    if (peepSpawnLoc.x == peep->NextLoc.x && peepSpawnLoc.y == peep->NextLoc.y)
```

Comment:
Are these different objects or could the equality be simplified?

Reply:
`peepSpawnLoc` is a CoordsXYZD and `peep->NextLoc` is a CoordsXYZ. Comparing them directly will also compare the Z coordinate, which is not supposed to happen.

Reply:
Ok, couldn't easily check on phone, thanks